apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/optimizeTc.ts

```diff
@@ -57,92 +78,213 @@ export function optimizeTc(
       if (f.operation === 'read' && f.path[0] === 'dyn') {
         const a = charTC.artifact.sets[f.path[1]]
         if (a) return constant(a)
-        if (!allSubstatKeys.includes(f.path[1] as any)) return constant(0)
+        if (!(allSubstatKeys as readonly string[]).includes(f.path[1]))
+          return constant(0)
```

Comment:
Not 100% sure what this does

Reply:
does this properly consider the configured set effects from `charTC`?

Reply:
Do we input those somewhere? If not, then this is one viable spot.

Reply:
Aren't we doing that one line above? If it's set counter, use `a`. If it's substat, make it variable, otherwise, set to 0.